packages/snjs/lib/services/item_manager.ts

```diff
@@ -734,14 +734,45 @@ export class ItemManager extends PureService {
         );
         const matchesQuery = regex.test(tag.title);
         const tagInNote = note
-          ? this.itemsReferencingItem(note.uuid).some((item) => item?.uuid === tag.uuid)
+          ? this.itemsReferencingItem(note.uuid).some(
+              (item) => item?.uuid === tag.uuid
+            )
           : false;
         return matchesQuery && !tagInNote;
       }),
       'title'
     );
   }
 
+  /**
+   * Returns all parents for a tag
+   * @param tag - The tag for which parents need to be found
+   * @returns Array containing all parent tags
+   */
+   public getTagParentChain(tag: SNTag): SNTag[] {
+    const delimiter = '.';
+    const tagComponents = tag.title.split(delimiter);
+    const parentTagsTitles: string[] = [];
+
+    const getImmediateParent = () => {
+      if (tagComponents.length > 1) {
+        // Remove last component
+        tagComponents.splice(-1, 1);
+        // Get immediate parent title and add it to chain
+        const immediateParentTitle = tagComponents.join(delimiter);
+        parentTagsTitles.push(immediateParentTitle);
+        // Get parent of this parent recursively
```

Comment:
I think the code is actually very readable and easy to follow so these comments aren't necessary :)